src/backends/perf_pt/decode.c

```diff
@@ -212,14 +213,15 @@ perf_pt_next_block(struct pt_block_decoder *decoder, int *decoder_status,
     }
     // It's possible at this point that we get notified of an event in the
     // stream. This will be handled in the next call to `perf_pt_next_block`.
-    assert((*decoder_status == 0) || (*decoder_status == pts_event_pending));
+    xassert(*decoder_status == 0 || *decoder_status == pts_event_pending,
+            "Unexpected decoder status: %d", *decoder_status);
 
     // XXX A truncated block occurs when a block straddles a section boundary.
     // In this case we may need some extra logic, but this should be rare.
-    assert(block.truncated == 0);
+    xassert(block.truncated == 0, "Truncated blocks are not implemented");
 
     // A block should have at least one instruction.
-    assert(block.ninsn > 0);
+    xassert(block.ninsn > 0, "Detected a block with 0 instructions");
```

Comment:
That apparently *can* happen, but I want to know when ;) So I guess that's challenging my assumption that it can't, so we can at a stretch call that an internal consistency check.